--- a/s3fs.install
+++ b/s3fs.install
@@ -249,9 +249,10 @@ function _s3fs_convert_table_to_binary_collation($table) {
       case 'mysql':
         // As stated here:
         // http://forums.mysql.com/read.php?103,19380,200971#msg-200971
         // MySQL doesn't directly support case sensitive UTF8 collation.
-        // Fortunately, 'utf8_bin' collation works for our purposes.
+        // Fortunately, 'utf8_bin' collation works for our purposes.
+        // TiDB doesn't support converting from utf8mb4 to utf8, so use utf8mb4_bin instead
         \Drupal::database()
-          ->query('ALTER TABLE {' . $table . '} CONVERT TO CHARACTER SET utf8 COLLATE utf8_bin');
+          ->query('ALTER TABLE {' . $table . '} CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_bin');
         break;
     }
     \Drupal::database()->schema()->addPrimaryKey($table, ['uri']);
